src/Stratis.FederatedPeg.Features.FederationGateway/Wallet/FederationWalletManager.cs

```diff
@@ -525,6 +525,25 @@ public bool RemoveTransaction(Transaction transaction)
             return updatedWallet;
         }
 
+        /// <inheritdoc />
+        public HashSet<(uint256, DateTimeOffset)> RemoveAllTransactions()
+        {
+            var removedTransactions = new HashSet<(uint256, DateTimeOffset)>();
+
+            lock (this.lockObject)
+            {
+                removedTransactions.UnionWith(this.Wallet.MultiSigAddress.Transactions.Select(t => (t.Id, t.CreationTime)));
```

Comment:
You don't need UnionWith here as you don't have many accounts. You can just have `removedTransactions = this.Wallet.MultiSigAddress.Transactions.Select(t => (t.Id, t.CreationTime));`